src/main/flight/interpolated_setpoint.c

```diff
@@ -28,103 +28,85 @@
 #include "fc/rc.h"
 #include "flight/interpolated_setpoint.h"
 
-static float projectedSetpoint[XYZ_AXIS_COUNT];
+static float setpointDeltaImpl[XYZ_AXIS_COUNT];
+static float prevSetpointDeltaImpl[XYZ_AXIS_COUNT];
+static float setpointDelta[XYZ_AXIS_COUNT];
+
+
 static float prevSetpointSpeed[XYZ_AXIS_COUNT];
+static float prevRcRawSetpoint[XYZ_AXIS_COUNT];
 static float prevRawSetpoint[XYZ_AXIS_COUNT];
-static float prevRawDeflection[XYZ_AXIS_COUNT];
-static uint16_t interpolationSteps[XYZ_AXIS_COUNT];
-static float setpointChangePerIteration[XYZ_AXIS_COUNT];
-static float deflectionChangePerIteration[XYZ_AXIS_COUNT];
-static float setpointReservoir[XYZ_AXIS_COUNT];
-static float deflectionReservoir[XYZ_AXIS_COUNT];
+static float prevSetpointAcceleration[XYZ_AXIS_COUNT];
+
 
 // Configuration
-static float ffLookaheadLimit;
-static float ffSpread;
 static float ffMaxRateLimit[XYZ_AXIS_COUNT];
 static float ffMaxRate[XYZ_AXIS_COUNT];
 
 void interpolatedSpInit(const pidProfile_t *pidProfile) {
     const float ffMaxRateScale = pidProfile->ff_max_rate_limit * 0.01f;
-    ffLookaheadLimit = pidProfile->ff_lookahead_limit * 0.0001f;
-    ffSpread = pidProfile->ff_spread;
     for (int i = 0; i < XYZ_AXIS_COUNT; i++) {
         ffMaxRate[i] = applyCurve(i, 1.0f);
         ffMaxRateLimit[i] = ffMaxRate[i] * ffMaxRateScale;
     }
 }
 
-FAST_CODE_NOINLINE float interpolatedSpApply(int axis, float pidFrequency, bool newRcFrame) {
-    const float rawSetpoint = getRawSetpoint(axis);
-    const float rawDeflection = getRawDeflection(axis);
+FAST_CODE_NOINLINE float interpolatedSpApply(int axis, bool newRcFrame, ffInterpolationType_t type) {
 
-    float pidSetpointDelta = 0.0f;
-    static int iterationsSinceLastUpdate[XYZ_AXIS_COUNT];
     if (newRcFrame) {
+        float rawSetpoint = getRawSetpoint(axis); 
+        
+        const float rxInterval = currentRxRefreshRate * 1e-6f;
```

Comment:
For missed frames we still get a new frame from the rx. It's just that two frames are repeated.

Reply:
The problem is that after missed frame, derivative will be about twice of normal value, but rxInterval stays the same ...

Reply:
That's true. There are actually several possible "missed frame" cases: 
- A measurement can be early on the TX relative to packet send rhythm, causing one frame to be delivered early and then one missed frame after the early one. Here derivative is twice for the first frame and then zero for the second. 
- A packet can just really go missing resulting in a zero derivative first and the 2x for the next frame.
- The tx sampling can move ahead or behind one tx frame consistently due to missing synchronization between sampling rate and tx frame rate.

The best we've been able to come up with here are the two mitigation features: ff_interpolated_averaged does a two point average on the ff and boost calculation. And ff_jerk_limit allows to limit how much boost jumps around due to these artificial changes in first derivative due to rc inconsistencies.

It works pretty well and doesn't create the level of delayed action as the previous version did.

Reply:
1) is TX bug, but this PR will use double value for both intervals (newRcFrame is false on repeated frame, so derivative is used twice)

2) can be handled by using 2*currentRxRefreshRate

3) is quite similar to 1)

Reply:
1.) Why would newRcFrame be false on a repeated frame?
2.) That would just cut ff in half, wouldn't it? We use the averaged feature to do two point averaging. It does help but doesn't completely solve the problem. I have experimented with all kinds of other approaches to fix the rc data stream, but this simple approach so far works best.

Maybe more in 4.2.

Reply:
1) Sorry, mixed it up with `isDuplicateFrame` (which seems unused)

2) Derivative will be about twice as large (assuming constant speed), this will divide it correctly ...

Reply:
Well spotted @ledvinap. `rcIsDuplicateFrame()` is part of the pre-existing code, but it is unused as well.